File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = [list(map(int, row.split(' ')))
                       for row in matrix_string.split('\n')]

    def row(self, index):
        return self.matrix[index-1]

    def column(self, index):
        return [rows[index-1] for rows in self.matrix]

File: typescript/matrix/matrix.ts
class Matrix {
  constructor(private matrix: string) {}

  get rows(): number[][] {
    return this.parseInputLines(this.matrix);
  }

  get columns(): number[][] {
    return this.transpose(this.rows);
  }

  private parseInputLines(matrix: string): number[][] {
    return matrix.split('\n').map((row) => row.split(' ').map(Number));
  }

  private transpose(matrix: number[][]): number[][] {
    return matrix[0].reduce(
      (columns, _, colIdx) => [...columns, matrix.map((row) => row[colIdx])],
      [] as number[][],
    );
  }
}

export default Matrix;
